### DISCUSSION TITLE
Connecting Palette 2s

### DISCUSSION THREAD

**Discussion by maoravi456** [xblax/flashforge_ad5m_klipper_mod#274] [Q&A]
Hey

I would like to connect Palette 2S to klipper, the issue is the USB
So, I thought on the idea of installing USBIP:
Host on pi which the Palette 2S connected to
Client on the 5MPro

then it will create virtual USB on the 5M - then I will be able to connect the Palette 2S

I couldn't figure out how to install USBIP package on 5M(apt does not work)..any idea?

**Comment by KaruroChori** [comment]
This is not a general purpose linux distribution, so there is no packet manager on it (mostly because there are strong limitations in the available memory).  
Adding packages is a manual process and requires a new image to be generated for the most part. It is based on *buildroot*, so you would have to be familiar with that.

**Reply by maoravi456** [reply]
ok thank you :)

do you have instructions on how to do it?

**Reply by KaruroChori** [reply]
https://github.com/xblax/flashforge_adm5_buildroot_chroot/tree/cleanup for what is used by this repo & https://buildroot.org/downloads/manual/manual.html would be your friend. But it is not going to be an easy ride :).

After that you have to package it in a release. There are probably some very early conversations on this repo covering that process as well.

**Comment by maoravi456** [comment]
So I've tried to understand how to connect Palette 2S
When I connect the device to my printer - it recognize it in the bus(can see it in lsusb) but no new folder appears(ttyUSB0/ttyACM0).
Then I connected the device to WSL2 and found out it connects via FTDI driver - which is missing from klippermod

There is a way to add FTDI driver to system? do I need to recompile? how can I do this?
https://www.intra2net.com/en/developer/libftdi/index.php
